src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -465,8 +465,33 @@ internal bool TryGetValue(int index, out DateTime value)
             ReadOnlySpan<byte> data = _utf8Json.Span;
             ReadOnlySpan<byte> segment = data.Slice(row.Location, row.SizeOrLength);
 
-            if (JsonHelpers.TryParseAsISO(segment, out DateTime tmp, out int bytesConsumed) &&
-                segment.Length == bytesConsumed)
+            DateTime tmp;
+            int bytesConsumed;
+
+            if (row.HasComplexChildren)
+            {
+                int backslash = segment.IndexOf(JsonConstants.BackSlash);
+                Debug.Assert(backslash != -1);
+
+                Span<byte> segmentUnescaped = stackalloc byte[segment.Length];
```

Comment:
The length needs to be validated as in-range before a stackalloc.

Reply:
For reference, the correct pattern would look something like this:

```cs
Span<byte> theSpan = ((uint)buffer.Length <= 128 /* or whatever const */)
    ? stackalloc byte[buffer.Length]
    : new byte[buffer.Length];
```